addons/stock/models/stock_picking.py

```diff
@@ -1345,18 +1345,27 @@ def _put_in_pack(self, move_line_ids, create_package_level=True):
                     ml.write(vals)
                     new_move_line.write({'product_uom_qty': done_to_keep})
                     move_lines_to_pack |= new_move_line
+            package_type = move_lines_to_pack.move_id.product_packaging_id.package_type_id
+            if len(package_type) == 1:
+                package.package_type_id = package_type
+            if len(move_lines_to_pack) == 1:
+                default_dest_location = move_lines_to_pack._get_default_dest_location()
+                move_lines_to_pack.location_dest_id = default_dest_location._get_putaway_strategy(
+                    product=move_lines_to_pack.product_id,
+                    quantity=move_lines_to_pack.product_uom_qty,
+                    package=package)
```

Comment:
Seems a bit too simple. It should be great to have something a bit better:
- If we have multiple`stock.move.line` they could go to the same location so the len(move_lines_to_pack) is too restrictive
- It's a bit too intrusive to modify the `stock.move.line` location after the put in pack without any warning. I would rather use a wizard asking that a more suitable location for the pack has been found and ask if he want to change. (for master only)
- Does the putaway strategy should still handle the products in the case of a package. Not sure or a least the package should have the priority I.M.O.

Reply:
For the first issue, the restriction is released after e58b24ff2790e150a09d6db87e492a11b0c19db3

Reply:
For the second issue. It's the whole point of the fix, if we move it to master, put_in_pack still can't use putaway rules in 15.0

Reply:
For third one, the priority is always package.package_type > packaging.package_type > product